ddtrace/internal/writer.py

```diff
@@ -286,6 +283,27 @@ def __init__(
             retry=tenacity.retry_if_exception_type((compat.httplib.HTTPException, OSError, IOError)),
         )
 
+    @staticmethod
+    def _infer_endpoint(url):
+        # type: (str) -> str
+        info = agent._get_info(url)
+        if info is None:
+            # The agent is not running
+            raise RuntimeError("The agent is not running")
+
+        if info == {}:
+            # The agent is running but does not have an info endpoint
+            for endpoint in ["/v0.%d/traces" % i for i in (4, 3)]:
+                if agent._test_endpoint(endpoint, url):
+                    return endpoint
+            raise RuntimeError("Failed to detect a valid traces endpoint")
+
+        endpoints = sorted([_ for _ in info.get("endpoints", []) if _.endswith("/traces")], reverse=True)
```

Comment:
this'll work up until `/v0.9/traces` but will start to break on `/v0.10/traces`. Not that we'll ever get that far... right? 🙄 

Hmm also what if the agent introduces a new endpoint that we don't have support for yet? Maybe better to just manually check for the supported endpoints